### PR TITLE
mrc-2619 Refresh backend dependencies

### PR DESCRIPTION
**Pull request by mwoodbri**
- Update Kotlin and Gradle to current versions
- Fix all resulting warnings
- Bump Fuel and Detekt to versions available from Maven Central, enabling us to drop the deprecated JCenter repo

### CODE REVIEW

File: buildkite/shared-build-env.dockerfile [hunk 1]
```diff
@@ -3,12 +3,12 @@ FROM vimc/node-docker-openjdk:master
 # Setup gradle
 COPY ./src/gradlew /mint/src/
 COPY ./src/gradle /mint/src/gradle/
+COPY ./src/settings.gradle /mint/src/
```
Comment (mwoodbri) [inline buildkite/shared-build-env.dockerfile:6]: Latest Gradle always requires a settings file

**Review by EmmaLRussell — COMMENTED** [review]
Should I need to do anything locally with gradle? - getting this error:
`Unable to find method 'org.gradle.api.artifacts.result.ComponentSelectionReason.getDescription()Ljava/lang/String;'.`

which persists after gradle/IDE restart

**Comment by mwoodbri** [comment]
> Unable to find method 'org.gradle.api.artifacts.result.ComponentSelectionReason.getDescription()Ljava/lang/String;'.

Ah... sorry. Sounds like you might be on an [older IntelliJ](https://stackoverflow.com/questions/67136909/gradle-issues-details-in-discription). Could you try this, and see whether Gradle v6 is compatible?
```shell
cd src
./gradlew wrapper --gradle-version 6.9.1
```

**Comment by EmmaLRussell** [comment]
> > Unable to find method 'org.gradle.api.artifacts.result.ComponentSelectionReason.getDescription()Ljava/lang/String;'.
> 
> Ah... sorry. Sounds like you might be on an [older IntelliJ](https://stackoverflow.com/questions/67136909/gradle-issues-details-in-discription). Could you try this, and see whether Gradle v6 is compatible?
> 
> ```shell
> cd src
> ./gradlew wrapper --gradle-version 6.9.1
> ```

Yes, you're right Thanks!

**Pull request merged**